Issue #7768: Update AbstractChecks to log DetailAST - AnnotationLocation

Malintha1996 opened the pull request:
This resolves #7768 

## site update:
![Capture](https://user-images.githubusercontent.com/35211477/76689926-00d45e00-6661-11ea-822b-ecdc4d60e4e4.PNG)

## diff file:
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/index.html
## Goal 
Update AbstractChecks to log DetailAST - AnnotationLocation

Malintha1996 commented:
> > diff file:
> 
> Waiting on difference file.

done

rnveach commented:
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/android-launcher/index.html
Configuration was not modified for the check you are working on. ThrowsCount is not the module you are working on. You must change config.

We are expecting differences in regression for this PR.

Malintha1996 commented:
> https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/android-launcher/index.html
> Configuration was not modified for the check you are working on. ThrowsCount is not the module you are working on. You must change config.
> 
> We are expecting differences in regression for this PR.

done

rnveach commented:
https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/checkstyle/index.html
> Number of unique base messages reported below: 115
Number of unique patch messages reported below: 121

Counts don't match up as was mentioned in the parent issue for this.

Please find the new violations that weren't being reported before and ensure we have them as directly part of our `AnnotationLocation` tests.

Malintha1996 commented:
> https://malintha1996.github.io/checkstyle-tester-reports/AnnotationLocation/diff/checkstyle/index.html
> 
> > Number of unique base messages reported below: 115
> > Number of unique patch messages reported below: 121
> 
> Counts don't match up as was mentioned in the parent issue for this.
> 
> Please find the new violations that weren't being reported before and ensure we have them as directly part of our `AnnotationLocation` tests.

The new violations are generated as annotations on the same line has been considered as different contexts(entities) in the AST. For example, when we have @Annotation1 @Annotation2 two messages for Annotation1 and Annotation 2 are created. But before, only one violation message was generated. I have already added this new violation messages to the assertions in the test cases.

rnveach commented:
>  I have already added this new violation messages to the assertion lists in the test cases.

Can you point me to which ones, I wasn't seeing them when I looked.

Malintha1996 commented:
> > I have already added this new violation messages to the assertion lists in the test cases.
> 
> Can you point me to which ones, I wasn't seeing them when I looked.



> > I have already added this new violation messages to the assertion lists in the test cases.
> 
> Can you point me to which ones, I wasn't seeing them when I looked.

#158,#159  #A161, #A162   #170,#171  #A173,#A174   #182,#183    #A185,#A186

Malintha1996 commented on src/test/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheckTest.java:
```diff
@@ -285,10 +285,13 @@ public void testAnnotationParameterized() throws Exception {
-            "25: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "15:5: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "17:5: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
+            "17:17: " + getCheckMessage(MSG_KEY_ANNOTATION_LOCATION_ALONE, "Annotation"),
```
@rnveach

rnveach replied:
Thanks. This is what I was looking for.
